core/src/main/cljc/jarl/builtins/http.cljc

```diff
@@ -3,14 +3,25 @@
             [clojure.set :as set]
             [clojure.string :as str]
             [clojure.walk :as walk]
+            [tick.core :as t]
             [jarl.builtins.utils :as utils]
             [jarl.exceptions :as errors]
             [jarl.encoding.json :as json]
             [jarl.encoding.yaml :as yaml]
-            [jarl.http-client :as http-client]))
+            [jarl.http-client :as http-client]
+            [jarl.time :as time]))
 
 (def ^:private default-req-timeout 5000)
 
+(def inter-query-cache (atom {}))
+
+(defn- force-cache? [req]
+  (and (:force-cache req)
+       (pos-int? (:force-cache-duration-seconds req))))
```

Comment:
See the docs on [atoms](https://clojure.org/reference/atoms):
> As with all reference types, the intended use of atom is to hold one of Clojure’s immutable data structures. And, similar to ref’s alter and agent’s send, you change the value by applying a function to the old value. This is done in an atomic manner by [swap!](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/swap!) Internally, [swap!](https://clojure.github.io/clojure/clojure.core-api.html#clojure.core/swap!) reads the current value, applies the function to it, and attempts to compare-and-set! it in. Since another thread may have changed the value in the intervening time, it may have to retry, and does so in a spin loop. The net effect is that the value will always be the result of the application of the supplied function to a current value, atomically. However, because the function might be called multiple times, it must be free of side effects.

Reply:
Indeed, on line 106 you'll find: 
```
; TODO: verify correct type of each provided value
```
We might want to use [malli](https://github.com/metosin/malli) for that, as I've started working on for the OPA configuration. But I'm leaving that for future PRs.